File: Programação Orientada a Objetos/Trabalho Prático 2/mavenproject1/src/main/java/Apartamento.java
public class Apartamento extends Imovel{
    //Características específicar do Apartamento
    private int andar;
    private float taxa_condominio;
    private boolean elevador;
    private boolean sacada;

    //Construtor
    public Apartamento (int i, float v, String p, String r, String b, String c, int n, int q, int ba, int a, float tc, boolean e, boolean s) {
        super(i, v, p, r, b, c, n, q, ba);
        andar = a;
        taxa_condominio = tc;
        elevador = e;
        sacada = s;
    }

    //Método do printar
    @Override
    public void print() {
        super.print();
        if (elevador == true)
            System.out.printf("Tem elevador\n");

    }

    //Getters e Setters
    public int getAndar() {
        return andar;
    }

    public void setAndar(int a){
        andar = a;
    }

    public float getTaxaCondominio() {
        return taxa_condominio;
    }

    public void setTaxaCondominio(float tc) {
        taxa_condominio = tc;
    }

    public boolean getElevador() {
        return elevador;
    }

    public void setElevador(boolean e){
        elevador = e;
    }

    public boolean getSacada() {
        return sacada;
    }

    public void setSacada(boolean s){
        sacada = s;
    }
}

File: Programação Orientada a Objetos/Trabalho Prático 1/apartamento.cpp
#include "apartamento.h"
#include "imovel.h"
 
//Construtor
Apartamento::Apartamento(int i, float v, std::string p, std::string r, std::string b, std::string c, int n, int q, int ba, int a, float tc, bool e, bool s) : Imovel(i, v, p, r, b, c, n, q, ba){
    andar = a;
    taxa_condominio = tc;
    elevador = e;
    sacada = s;
}

//Overload do método print
std::ostream& Apartamento::print(std::ostream& os) const{
    os  << "----\n"
        << "\t" << this->getProprietario() << "\n"
        << "\t" << this->getValor() << "\n"
        << "\t" << this->getQuartos() << "\n"
        << "\t" << this->getRua() << "\n"
        << "\t" << this->getBairro() << "\n"
        << "\t" << this->getCidade() << "\n"
        << "\t" << ((this->getElevador()) ? "Possui elevador" : "Não possui elevador") << "\n"
        << "----\n";
    return os;
}

//Getters e Setters
int Apartamento::getAndar() const{
    return andar;
}

void Apartamento::setAndar(int a){
    andar = a;
}

float Apartamento::getTaxaCondominio() const{
    return taxa_condominio;
}

void Apartamento::setTaxaCondominio(float tc){
    taxa_condominio = tc;
}

bool Apartamento::getElevador() const{
    return elevador;
}

void Apartamento::setElevador(bool e){
    elevador = e;
}

bool Apartamento::getSacada() const{
    return sacada;
}

void Apartamento::setSacada(bool s){
    sacada = s;
}